dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/data/ReturnTypesResolver.java

```diff
@@ -84,15 +90,16 @@ private Optional<TypeName> orDefaultDeserializerFactory(
                 context.reportError("When returning raw Response, remember to add @MustBeClosed annotation", element);
                 return Optional.empty();
             }
-            return Optional.of(context.getTypeName(ResponseDeserializer.class));
+            return Optional.of(CodeBlock.of("new $T()", context.getTypeName(ResponseDeserializer.class)));
         } else if (isInputStreamType(returnType)) {
             if (!hasMustBeClosed) {
                 context.reportError("When returning InputStream, remember to add @MustBeClosed annotation", element);
                 return Optional.empty();
             }
-            return Optional.of(context.getTypeName(InputStreamDeserializer.class));
+            return Optional.of(
+                    CodeBlock.of("new $T(runtime.bodySerDe())", context.getTypeName(InputStreamDeserializer.class)));
         }
-        return Optional.of(context.getTypeName(Json.class));
+        return Optional.of(CodeBlock.of("new $T(runtime.bodySerDe())", context.getTypeName(Json.class)));
```

Comment:
nit: Can we add constructor overloads that take `ConjureRuntime` so that instantiation of the built-in deserializer factories is consistent with custom ones?


```suggestion
        return Optional.of(CodeBlock.of("new $T(runtime)", context.getTypeName(Json.class)));
```